### PR TITLE
- Added STS support for ec2.py for handling multiple accounts

### PR DESCRIPTION
**Pull request by linuxbsdfreak**
##### SUMMARY
Following is the process to scan multiple accounts via STS:

Step 1: Create a role in the destination account from which we would like to scan the instances from and authorize the user and the account from the source account to AssumeRole
`aws iam create-role --path /service/ --role-name cross-access-role --assume-role-policy-document '{ "Version": "2008-10-17", "Statement": [ { "Sid": "", "Effect": "Allow", "Principal": { "AWS": "arn:aws:iam:::user/" }, "Action": "sts:AssumeRole" } ] }'`

Step 2: Attach a policy to the role to only list instances and describe tags
`aws iam put-role-policy --policy-name 'cross-access-policy' --role-name cross-access-role --policy-document '{ "Statement": [ { "Sid": "", "Action": [ "ec2:DescribeInstances", "ec2:DescribeTags", "route53:Get", "route53:List", "rds:Describe", "elasticloadbalancing:Describe" ], "Effect": "Allow", "Resource": "*" } ] }'`

Step3: `export AWS_ACCESS_KEY_ID=xxxx; export AWS_SECRET_ACCESS_KEY=xxxxxx` for the user present in the master account who has access to perform the cross account access

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
plugins/inventory/ec2.ini
plugins/inventory/ec2.py

##### ANSIBLE VERSION
2.3

### CODE REVIEW

File: plugins/inventory/ec2.ini [hunk 1]
```diff
@@ -42,10 +42,10 @@ vpc_destination_variable = ip_address
 
 # To tag instances on EC2 with the resource records that point to them from
 # Route53, uncomment and set 'route53' to True.
-route53 = False
+route53 = True
 
 # To exclude RDS instances from the inventory, uncomment and set to False.
-#rds = False
```
Comment (bcoca) [inline plugins/inventory/ec2.ini:48]: why change the defaults?

**Comment by linuxbsdfreak** [comment]
You can disable the values. I had enabled the config just to scan the RDS instances.

**Comment by amenonsen** [comment]
Just a quick note to say that there are other PRs that address the same need (to have `ec2.py` use multiple accounts).

**Comment by xenithorb** [comment]
Any movement on this? I need this to work on a client's account

**Comment by s-hertel** [comment]
Hi, @linuxbsdfreak. STS credentials are supported by boto3 and the new [aws_ec2 inventory plugin](https://docs.ansible.com/ansible/latest/plugins/inventory/aws_ec2.html). You can set up your credential files to do this with https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#assume-role-provider. It doesn't look like this will rebase cleanly since features have since been added to the contrib scripts that won't be assuming roles correctly and we aren't actively working on the contrib scripts. Feel free to reach out if you have questions.

**Pull request closed without merging**